src/commands/extract.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import * as Constants from '../constants';
+import { ApiTreeItem } from '../explorer/ApiTreeItem';
+import { ServiceTreeItem } from '../explorer/ServiceTreeItem';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { cpUtils } from '../utils/cpUtils';
+import { dotnetUtils } from '../utils/dotnetUtils';
+
+export async function extractService(node?: ServiceTreeItem): Promise<void> {
+    if (!node) {
+        node = <ServiceTreeItem>await ext.tree.showTreeItemPicker(ServiceTreeItem.contextValue);
+    }
+
+    const uris = await askFolder();
+    const templatesFolder = await createTemplatesFolder(uris);
+    const sourceApimName = node.root.serviceName;
+    const resourceGroup = node.root.resourceGroupName;
+
+    //generate linked templates
+    const extractConfig = generateExtractConfig(sourceApimName, resourceGroup, templatesFolder);
+    const configFile = path.join(uris[0].fsPath, `${sourceApimName}.json`);
+
+    await fse.writeFile(configFile, extractConfig);
+
+    window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("Extracting", `Extracting service '${sourceApimName}' to '${templatesFolder}'`),
+            cancellable: false
+        },
+        async () => {
+            await dotnetUtils.checkDotnetInstalled();
+            await runExtractor(configFile);
+        }
+    ).then(
+        () => {
+            window.showInformationMessage(localize("Extracted", `Extraction completed!`));
+        });
+}
+
+export async function extractAPI(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    const uris = await askFolder();
+    const templatesFolder = await createTemplatesFolder(uris);
+    const sourceApimName = node.root.serviceName;
+    const resourceGroup = node.root.resourceGroupName;
+    const apiName = node.apiContract.name == null ? "" : node.apiContract.name;
```

Comment:
just use node.apiContract.name

Reply:
node.apiContract.name is a property, it can be only string or undefined, (undefined == false)  is false, so I have to do (node.apiContract.name == null ?) or (node.apiContract.name === undefined ?)